### ISSUE TITLE
Ability to configure ephemeral resource limits on pods

### ISSUE THREAD

**Issue by FleetAdmiralButter** [uselagoon/build-deploy-tool#198]
Hi,

We've recently run into an issue where a project deployed on Lagoon was able to cause a worker node to stop working properly by filling up the `/tmp` folder with large files. 

This caused the root filesystem on the node to fill up and it went into DiskPressure, causing pods to get evicted and rescheduled.

Would you consider adding a feature to Lagoon where deployed pods have `spec.containers[].resources.limits.ephemeral-storage` configured with a default value?

Thanks!

**Comment by smlx** [comment]
AFAIK this is a k8s v1.25+ feature https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#local-ephemeral-storage

**Comment by shreddedbacon** [comment]
> AFAIK this is a k8s v1.25+ feature https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#local-ephemeral-storage

Ah thanks, I just read this too. So the feature might be delayed then.

**Comment by FleetAdmiralButter** [comment]
FWIW, I tested this on EKS 1.22 and 1.24, with a limit of 100Mi and my pod was indeed terminated once I exceeded that:

![image](https://user-images.githubusercontent.com/37036084/227412216-71bd4419-3965-4c4f-a298-619ed22497ab.png)

**Comment by shreddedbacon** [comment]
Thanks for checking that. I might see if I can sneak this in real quick to the next release then. It is pretty low effort.

**Comment by shreddedbacon** [comment]
I've been playing around with this, and it seems if you set the limit, the requests defaults to the same value unless you've already defined the requests as a lower value. In Lagoon, we don't set a value currently.

I extended the PR I'm working on to support also setting the requests value too, because of the following:

My concerns with exposing this is [scheduling pods](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#how-pods-with-ephemeral-storage-requests-are-scheduled). Since each node will have a [certain amount of ephemeral storage it can allocate](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable). Since Lagoon currently doesn't set a requests value for `ephemeral-storage` it is just ignored by the scheduler.

Playing around in my lab, I get the following when deploying an environment that had the limit set to a value I wanted, without also setting the requests (in this case, my nodes have 120Gi allocatable ephemeral-storage, so I set just the limit to 120Gi) once the pods started to roll out, some pods in this environment didn't schedule because of insufficient allocatable ephemeral-storage on the nodes.
```
0/5 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: true}, that the pod didn't tolerate, 4 Insufficient ephemeral-storage.
```

TLDR; setting a limit when no requests is set, will also sets the requests value to be the same as the limit. This can impact how pods with requests are scheduled unless you also set the requests to a low value you're willing to accept, or 0.

**Comment by FleetAdmiralButter** [comment]
Thanks for this @shreddedbacon! Very much appreciated.
